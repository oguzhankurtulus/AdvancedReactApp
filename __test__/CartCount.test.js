import { render } from '@testing-library/react';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
    it('Renders', () => {
        render(<CartCount count={10} />);
    });

    it('Matches snapshot', () => {
        const { container } = render(<CartCount count={11} />);
        expect(container).toMatchSnapshot();
    });

    it('Updates via props', () => {
        const { container, rerender } = render(<CartCount count={11} />);
        expect(container).toHaveTextContent('11');
        rerender(<CartCount count={12} />);
        expect(container).toHaveTextContent('12');
        expect(container).toMatchSnapshot();
    });
});
