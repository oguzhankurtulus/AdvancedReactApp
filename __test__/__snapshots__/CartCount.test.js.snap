// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count count-enter count-enter-active"
      >
        12
      </div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count count-exit count-exit-active"
      >
        11
      </div>
    </div>
  </span>
</div>
`;
